### ISSUE TITLE
Quick Start Guide Confusion

### ISSUE THREAD

**Issue by abhillman** [yesodweb/yesod#1368]
Thank you kindly for all your hard work on this web framework.The quickstart guide (http://www.yesodweb.com/page/quickstart) suggests the following:

```
stack build yesod-bin cabal-install --install-ghc
```

This goes and downloads and installs plethora of libraries that seem to already be present on my system when from both `haskell-platform` and from running `stack new my-project yesod-sqlite`. As a relative newcomer to Haskell, this process feels confusing and raises questions like:

* Will I have to do this every time I want to create a Yesod project?
* Why are we re-installing GHC?
* How are these new dependencies access if they don't "interfere with any system-level installation"?
* Is this best-practice? Are there other ways to do this that might be faster, yet are still a best-practice?

Could it be sensible to update the quick start documentation to give a clearer explanation about what's going on? In the long term, could it be possible to reduce the amount of time it takes to get started?

**Comment by snoyberg** [comment]
I don't believe such content is appropriate for a quick start guide which, by its nature, is intended to give straightforward instructions on how to do something, not the philosophy behind that choice. I will say that the Haskell Platform itself is a major problem, and we do a lot to avoid using it anywhere or recommend its usage.

As to your direct questions:

* You will not have to reinstall GHC and the libraries each time you create a Yesod project, unless a new GHC version is being used
* We reinstall GHC because we can't trust the system GHC. It's been a long source of bugs for people, due to such things as missing profiling libraries
* Building all the libraries takes time
